s2i/assemble

```diff
@@ -13,6 +13,26 @@ function finish {
 }
 trap finish EXIT
 
+echo "---> Understand root folder for source code"
+if [ -z "${ROOT_PACKAGE}" ]
+then
+    if [ ! -z "${SOURCE_REPOSITORY_URL}" ]
+    then
+        ROOT_PACKAGE=$(echo "$SOURCE_REPOSITORY_URL" | sed -re "s/(https:\/\/|git@)([^?]+)\.git/\2/g" | sed -re "s/:/\//g")
+        if [ ! -z "${CONTEXT_DIR}" ]
+        then
+            ROOT_PACKAGE="$ROOT_PACKAGE/$CONTEXT_DIR"
+        fi
+    fi
+fi
+
+if [ -z "${ROOT_PACKAGE}" ]
```

Comment:
This looks like it can be moved up to an "else" clause of the `SOURCE_REPOSITORY_URL` check